packages/react-inbox/src/components/Messages2.0/Message.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { ReactNode, useMemo } from "react";
+import classNames from "classnames";
+import { IActionBlock, ITextBlock } from "@trycourier/react-provider";
+import { useInbox } from "@trycourier/react-hooks";
+
+import { TextBlock, getIcon, TimeAgo, Title } from "../Message/styled";
+import { InboxProps } from "../../types";
+
+import { getTimeAgo } from "~/lib";
+import Markdown from "markdown-to-jsx";
+
+import deepExtend from "deep-extend";
+import styled from "styled-components";
+import useMessageOptions, { IMessageOption } from "~/hooks/use-message-options";
+import OptionsDropdown from "../OptionsDropdown";
+
+export interface IMessageProps {
+  blocks?: Array<ITextBlock | IActionBlock>;
+  created: string;
+  icon?: string;
+  messageId: string;
+  read?: boolean;
+  title: string;
+  unread?: number;
+  data?: {
+    clickAction: string;
+  };
+  trackingIds?: {
+    clickTrackingId: string;
+    deliverTrackingId: string;
+    readTrackingId: string;
+    unreadTrackingId: string;
+  };
+}
+
+const containerStyles = ({ theme }) =>
+  deepExtend(
+    {
+      display: "flex",
+      position: "relative",
+      padding: "10px",
+      backgroundColor: "#F9FAFB",
+      alignItems: "center",
+      borderBottom: "1px solid rgba(203,213,224,.5)",
+      "&.read": {
+        filter: "grayscale(60%)",
+        background: "#F2F6F9",
+      },
+      "&:not(.read):hover": {
+        background:
+          "linear-gradient(180deg, rgba(33, 150, 243, 0.24) 0%, rgba(33, 150, 243, 0.096) 100%)",
+      },
+    },
+    theme.message?.container ?? {}
+  );
+
+const DefaultContainer = styled.div(containerStyles);
+const ClickableContainer = styled.a(containerStyles);
+
+const Contents = styled.a(({ theme }) => ({
+  marginRight: "auto",
+  textAlign: "left",
+  ...theme.message?.contents,
+}));
+
+const UnreadIndicator = styled.div(({ theme }) =>
+  deepExtend(
+    {
+      height: "100%",
+      width: 2,
+      backgroundColor: theme?.brand?.colors?.primary ?? "#9121c2",
+      position: "absolute",
+      left: "1px",
+    },
+    theme?.message?.unreadIndicator
+  )
+);
+const Message: React.FunctionComponent<{
+  formattedTime?: string;
+  read?: boolean;
+  renderedIcon: ReactNode;
+  renderTextBlock: React.FunctionComponent<ITextBlock> | undefined;
+  textBlocks: ITextBlock[];
+  title?: string;
+  messageOptions: IMessageOption[];
+}> = ({
+  read,
+  formattedTime,
+  renderedIcon,
+  renderTextBlock,
+  textBlocks,
+  title,
+  messageOptions,
+}) => {
+  return (
+    <>
+      {!read && <UnreadIndicator />}
+      {renderedIcon}
+      <Contents>
+        <Title>{title}</Title>
+        {textBlocks?.map((block: ITextBlock, index: number) => {
+          if (renderTextBlock) {
+            const Block = renderTextBlock;
+            return <Block {...block} key={index} />;
+          }
+
+          return (
+            <TextBlock key={index} data-testid="message-body">
```

Comment:
One small note, using index as a key can cause weird [rendering bugs and performance issues](https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/). Probably not an issue here but something to keep an eye on.